File: doc/src/scripts/tutorials/reference-frames/VisualDebugging.java
import krpc.client.Connection;
import krpc.client.RPCException;
import krpc.client.services.SpaceCenter;
import krpc.client.services.SpaceCenter.ReferenceFrame;
import krpc.client.services.Drawing;

import org.javatuples.Triplet;

import java.io.IOException;

public class VisualDebugging {
    public static void main(String[] args) throws IOException, RPCException {
        Connection connection = Connection.newInstance("Visual Debugging");
        SpaceCenter spaceCenter = SpaceCenter.newInstance(connection);
        Drawing drawing = Drawing.newInstance(connection);
        SpaceCenter.Vessel vessel = spaceCenter.getActiveVessel();

        ReferenceFrame refFrame = vessel.getSurfaceVelocityReferenceFrame();
        drawing.addDirection(
          new Triplet<Double, Double, Double>(0.0, 1.0, 0.0), refFrame, 10, true);
        while (true) {
        }
    }
}

File: doc/src/scripts/tutorials/reference-frames/VisualDebugging.cpp
#include <krpc.hpp>
#include <krpc/services/space_center.hpp>
#include <krpc/services/ui.hpp>
#include <krpc/services/drawing.hpp>

int main() {
  krpc::Client conn = krpc::connect("Visual Debugging");
  krpc::services::SpaceCenter space_center(&conn);
  krpc::services::Drawing drawing(&conn);
  auto vessel = space_center.active_vessel();

  auto ref_frame = vessel.surface_velocity_reference_frame();
  drawing.add_direction(std::make_tuple(0, 1, 0), ref_frame);
  while (true) {
  }
}
